Cython/Utility/StringTools.c

```diff
@@ -113,6 +113,14 @@ static int __Pyx_PyUnicodeBufferContainsUCS4_BMP(Py_UNICODE* buffer, Py_ssize_t
 #endif
 
 static CYTHON_INLINE int __Pyx_UnicodeContainsUCS4(PyObject* unicode, Py_UCS4 character) {
+#if CYTHON_COMPILING_IN_LIMITED_API
+    // Note that from Python 3.7, the indices of FindChar account for wraparound so no
+    // need to check the length
+    Py_ssize_t idx = PyUnicode_FindChar(unicode, character, 0, -1, 1);
```

Comment:
You could also pass `PY_SSIZE_T_MAX` as end to avoid the risk of an off-by-one error for `-1` (because, is that inclusive or exclusive?).

Given that there's a C-API function for this, though, I wonder if we're really still faster than that. The implementation seems quite efficient. From my expecience, `PyUnicode_READ()` isn't the fastest thing to use, compared to a straight memory loop.

Reply:
I've taken this to mean "just use PyUnicode_FindChar in all cases". Happy to revert if this wasn't what you mean though!